### PR TITLE
Import documents: async url reader

### PR DESCRIPTION
**Pull request by PrimozGodec**
##### Issue

Import documents downloads files synchronously which is slower to asynchronous download.

##### Description of changes
Implement asynchronous download with HTTPX library.

Speedups:
predlogi-vladi-1k: 670s-> 47s
predlogi-vladi: now 540s for all documents (before more than 1h30m)

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -4,6 +4,7 @@ biopython # Enables Pubmed widget.
 conllu
 docx2txt>=0.6
 gensim>=4.1.2  # https://github.com/RaRe-Technologies/gensim/issues/3226
+httpx
```
Comment (PrimozGodec) [inline requirements.txt:7]: We are actually not adding a new dependency in the environment since Orange already depends on HTTPX.

**Comment by VesnaT** [comment]
Can you please rebase?

**Comment by PrimozGodec** [comment]
> Can you please rebase?

Done

File: orangecontrib/text/import_documents.py [hunk 2]
```diff
@@ -1,20 +1,21 @@
-import contextlib
-import datetime
+import asyncio
 import fnmatch
 import logging
 import os
 import pathlib
 import re
+from urllib.parse import urlparse
```
Comment (VesnaT) [inline orangecontrib/text/import_documents.py:7]: This is never used.

**Review by VesnaT — COMMENTED** [review]
It works great. If you remove the unused import, I'll merge.

File: orangecontrib/text/import_documents.py (hunk 2 above)
Comment (PrimozGodec) [reply orangecontrib/text/import_documents.py]: Ups. :)

**Comment by PrimozGodec** [comment]
> It works great. If you remove the unused import, I'll merge.

Done

**Pull request merged**